### PR TITLE
Test - send a value transaction to local Tangle

### LINKED ISSUE #300: Client - update send_message to v2
## Description

The send message endpoint has been changed
- update restful API route to v2
- update `core_message_sign_transaction` method
- `test_send_core_message_tx` tests should be updated and passed

### PR DESCRIPTION
**Pull request by oopsmonk**
# Description of change

fixes #300, sending value transactions on local private Tangle.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bugfix (a non-breaking change which fixes an issue)

## How the change has been tested

value tx is tested via test_send_msg_tx_extended

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/client/api_restful/test_send_message.c [hunk 1]
```diff
@@ -4,13 +4,22 @@
 #include <stdio.h>
 #include <unity/unity.h>
```
Comment (matjazv) [inline tests/client/api_restful/test_send_message.c]: `unity.h` is not a system header file.

File: tests/client/api_restful/test_send_message.c [hunk 2]
```diff
@@ -175,24 +184,129 @@ void test_send_core_message_tagged_data_tag_max_len() {
...
+        total_balance += o->amount;
+        // check balane
+        if (total_balance >= send_amount) {
+          // add it to tx payload and break
+          tx_payload_add_input(tx, 0, output_res->u.data->tx_id, output_res->u.data->output_index, &sender_key);
+          get_output_response_free(output_res);
+          break;
+        } else {
+          // add it to tx payload and fetch next output data
+          tx_payload_add_input(tx, 0, output_res->u.data->tx_id, output_res->u.data->output_index, &sender_key);
```
Comment (matjazv) [inline tests/client/api_restful/test_send_message.c]: `tx_payload_add_input(tx, 0, output_res->u.data->tx_id, output_res->u.data->output_index, &sender_key);` can be extracted from `if `and `else` blocks and moved above them.

File: tests/client/api_restful/test_send_message.c [hunk 3]
```diff
@@ -175,24 +184,129 @@ void test_send_core_message_tagged_data_tag_max_len() {
...
+  uint64_t total_balance = 0;
+  for (size_t i = 0; i < res_outputs_output_id_count(res); i++) {
+    res_output_t* output_res = get_output_response_new();
+    printf("fetch output: %s\n", res_outputs_output_id(res, i));
+    TEST_ASSERT(get_output(&ctx, res_outputs_output_id(res, i), output_res) == 0);
+    if (!output_res->is_error) {
+      if (output_res->u.data->output->output_type == OUTPUT_EXTENDED) {
+        output_extended_t* o = (output_extended_t*)output_res->u.data->output->output;
+        total_balance += o->amount;
+        // check balane
```
Comment (matjazv) [inline tests/client/api_restful/test_send_message.c]:
```suggestion
        // check balance
```

**Review by matjazv — CHANGES_REQUESTED** [review]
Nice. Client C library is able to send a value transactions to the tangle.

File: tests/client/api_restful/test_send_message.c (hunk 2 above)
Comment (oopsmonk) [reply tests/client/api_restful/test_send_message.c]: nice catch :)

**Review by rebinmannady93 — APPROVED** [review]

File: tests/client/api_restful/test_send_message.c [hunk 4]
```diff
@@ -175,24 +184,127 @@ void test_send_core_message_tagged_data_tag_max_len() {
...
+  address_to_bech32(&addr_send, hrp, bech32_sender, sizeof(bech32_sender));
+  address_to_bech32(&addr_recv, hrp, bech32_receiver, sizeof(bech32_receiver));
+  printf("sender: %s\nreceiver: %s\n", bech32_sender, bech32_receiver);
+
+  transaction_payload_t* tx = tx_payload_new();
+  TEST_ASSERT_NOT_NULL(tx);
+
+  // get outputs from an address
+  uint64_t send_amount = 1000000;  // send out 1Mi
+  res_outputs_id_t* res = res_outputs_new();
```
Comment (rebinmannady93) [inline tests/client/api_restful/test_send_message.c:219]:
```suggestion
  res_outputs_id_t* res = res_outputs_new();
  TEST_ASSERT_NOT_NULL(res);
```

File: tests/client/api_restful/test_send_message.c [hunk 5]
```diff
@@ -175,24 +184,127 @@ void test_send_core_message_tagged_data_tag_max_len() {
...
+  outputs_query_list_t* query_param = outputs_query_list_new();
```
Comment (rebinmannady93) [inline tests/client/api_restful/test_send_message.c:221]:
```suggestion
  outputs_query_list_t* query_param = outputs_query_list_new();
  TEST_ASSERT_NOT_NULL(query_param);
```

**Review by rebinmannady93 — CHANGES_REQUESTED** [review]

**Review by matjazv — APPROVED** [review]

**Review by rebinmannady93 — APPROVED** [review]

**Pull request merged**